src/vs/workbench/contrib/webview/browser/pre/service-worker.js

```diff
@@ -318,6 +318,17 @@ async function processResourceRequest(event, requestUrlComponents) {
 			headers['Last-Modified'] = new Date(entry.mtime).toUTCString();
 		}
 
+		// support COI requests, see network.ts#COI.getHeadersFromQuery(...)
+		const coiRequest = new URL(event.request.url).searchParams.get('vscode-coi');
+		if (coiRequest === '3') {
+			headers['Cross-Origin-Opener-Policy'] = 'same-origin';
+			headers['Cross-Origin-Embedder-Policy'] = 'require-corp';
+		} else if (coiRequest === '2') {
+			headers['Cross-Origin-Embedder-Policy'] = 'require-corp';
+		} else if (coiRequest === '1') {
+			headers['Cross-Origin-Opener-Policy'] = 'same-origin';
+		}
```

Comment:
@mjbvz Can you help me here? The contract is that sub-resources, like worker and iframes, detect they are started from with a COI context and therefore want to inherit it. They do this via a `?vsocde-coi`-url-query. This works in web and works desktop resources, however I struggle with webviews... Interpreting the query here nor in `webviewProtocolProvider` seems to have an affect.